### PR TITLE
De-macrofiy precord.h, simplify precord copying code

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: src/precord.h [hunk 1]
```diff
@@ -28,7 +31,53 @@
...
+
+/****************************************************************************
+**
+*F  CAPACITY_PREC(<list>) . . . . . . . . . . . .  capacity of a plain record
+**
+**  'CAPACITY_PREC' returns the maximum capacity of a PREC.
+**
+*/
+static inline Int CAPACITY_PREC(Obj rec)
+{
```
Comment (ChrisJefferson) [inline src/precord.h]: Missing assert IS_PREC_OR_COMOBJ?

**Comment by ChrisJefferson** [comment]
I don't really like the `GVar` and `RNam` typedefs, I'd prefer just `UInt`s unless we hope one day to actually make them into a different type, but that might be a more general discussion.

File: src/precord.h (hunk 1 above)
Comment (fingolfin) [reply src/precord.h]: I imitated `CAPACITY_PLIST` here, which also does not contain that check -- all callers to this function already know they are dealing with a prec, too.

**Comment by fingolfin** [comment]
I dropped `GVar` and `RNam`.

**Pull request merged**